tests/csapi/apidoc_search_test.go

```diff
@@ -242,20 +242,32 @@ func TestSearch(t *testing.T) {
 
 			resp := alice.MustDoFunc(t, "POST", []string{"_matrix", "client", "v3", "search"}, searchRequest)
 			sce := "search_categories.room_events"
-			result0 := sce + ".results.0.result"
-			result1 := sce + ".results.1.result"
+
+			expectedEvents := map[string]string{
+				eventBeforeUpgrade: "Message before upgrade",
+				eventAfterUpgrade: "Message after upgrade",
+			}
 			must.MatchResponse(t, resp, match.HTTPResponse{
 				StatusCode: http.StatusOK,
 				JSON: []match.JSON{
-					match.JSONKeyPresent(sce + ".count"),
-					match.JSONKeyPresent(sce + ".results"),
 					match.JSONKeyEqual(sce+".count", float64(2)),
-					match.JSONKeyPresent(result0 + ".content"),
-					match.JSONKeyPresent(result0 + ".type"),
-					match.JSONKeyEqual(result0+".event_id", eventBeforeUpgrade),
-					match.JSONKeyEqual(result1+".event_id", eventAfterUpgrade),
-					match.JSONKeyEqual(result0+".content.body", "Message before upgrade"),
-					match.JSONKeyEqual(result1+".content.body", "Message after upgrade"),
+					match.JSONKeyArrayOfSize(sce+".results", 2),
+
+					// the results can be in either order: check that both are there and that the content is as expected
+					match.JSONCheckOff(sce+".results", []interface{} {eventBeforeUpgrade, eventAfterUpgrade}, func(res gjson.Result) interface{} {
+						return res.Get("event_id")
```

Comment:
```suggestion
						return res.Get("event_id").Str
```

Reply:
This is the bug. The error message on CI is unhelpful because it's using `%s` not `%v`: https://github.com/matrix-org/complement/blob/41787c72bf6c9faed27c840a0473426eaa4f6e77/internal/match/json.go#L122